### DISCUSSION TITLE
Get text font and font size of freetext annotations

### DISCUSSION THREAD

**Discussion by dantard** [pymupdf/PyMuPDF#2435] [Looking for help]
Hello, I'm looking how to get font and font size of freetext annotations when it is obtained through `page.annots()`.

Even though it is easy to set the font and the font size when it is added with `page.add_freetext_annot()` I haven't found the way of getting this information back.

Thanks in advance.

**Comment by JorjMcKie** [comment]
Sorry, there is no high-level support for this currently. You need to work your way to this info using low-level API:
The annotation appearance is made by the annotation's `/AP` object - ususally the subobject `/AP/N`. This object has a `/Resources` object inside which you should stumble over the object `/Font`. Example:
```python
doc.xref_get_key(a.xref, "AP/N/Resources/Font")
('dict', '<</Helv 7 0 R>>')
# may have multiple entries. Here we only have one:
print(doc.xref_object(7))
<<
  /Type /Font
  /Subtype /Type1
  /BaseFont /Helvetica
  /Encoding /WinAnsiEncoding
>>
```

**Reply by dantard** [reply]
I see, thank you.
I assume the size can be obtained using the `/DA` object?

**Reply by JorjMcKie** [reply]
Exactly!
